textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/field/SpanTuple.java

```diff
@@ -0,0 +1,39 @@
+package edu.uci.ics.textdb.common.field;
+
+import java.util.List;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IField;
+
+public class SpanTuple extends DataTuple{
+    
+    private IField spanField;
+    private Span span;
+    private String key;
+
+    public SpanTuple(List<Attribute> schema, IField[] fields, 

```

Comment:
@JavierJia .. I get your point. The reason we didn't use `SpanTupleWrapper` is we don't want to change the return type of `getNextTuple()` method of `DictionaryMatcher` from the existing `ITuple'. I will include the span information in schema, but I will also extra methods to get the start position, end position, span text etc.,